miniapps/multidomain/multidomain.cpp

```diff
@@ -0,0 +1,249 @@
+   block_wall_attributes[3] = 1;
+
+   Array<int> outer_cylinder_wall_attributes(
+      block_submesh.bdr_attributes.Max());
+   outer_cylinder_wall_attributes = 0;
+   outer_cylinder_wall_attributes[8] = 1;
+
+   h1_fes_block.GetEssentialTrueDofs(block_wall_attributes, ess_tdofs);
+
+   ConvectionDiffusionTDO d_tdo(h1_fes_block, ess_tdofs, 0.0, 1.0);
+
+   ParGridFunction temperature_block_gf(&h1_fes_block);
+   temperature_block_gf = 0.0;
+   ConstantCoefficient one(1.0);
+   temperature_block_gf.ProjectBdrCoefficient(one, block_wall_attributes);
+   Vector temperature_block;
+   temperature_block_gf.GetTrueDofs(temperature_block);
+
+   RK3SSPSolver d_ode_solver;
+   d_ode_solver.Init(d_tdo);
+
+   Array<int> cylinder_surface_attributes(1);
+   cylinder_surface_attributes[0] = 9;
+
+   auto cylinder_surface_submesh = ParSubMesh::CreateFromBoundary(mesh,
+                                                                  cylinder_surface_attributes);
+
+   ParFiniteElementSpace cylinder_surface_fes(&cylinder_surface_submesh, &h1_fec);
+   ParGridFunction cylinder_surface_gf(&cylinder_surface_fes);
+
+   double dt = 1.0e-5;
+   double t_final = 5.0;
+   double t = 0.0;
+   bool last_step = false;
+   int vis_steps = 10;
+
+   char vishost[] = "128.15.198.77";
+   int  visport   = 19916;
+   socketstream cyl_sol_sock(vishost, visport);
+   cyl_sol_sock << "parallel " << num_procs << " " << myid << "\n";
+   cyl_sol_sock.precision(8);
+   cyl_sol_sock << "solution\n" << cylinder_submesh << temperature_cylinder_gf <<
+                "pause\n" << std::flush;
+
+   socketstream block_sol_sock(vishost, visport);
+   block_sol_sock << "parallel " << num_procs << " " << myid << "\n";
+   block_sol_sock.precision(8);
+   block_sol_sock << "solution\n" << block_submesh << temperature_block_gf <<
+                  "pause\n" << std::flush;
+
+   for (int ti = 1; !last_step; ti++)
+   {
+      if (t + dt >= t_final - dt/2)
+      {
+         last_step = true;
+      }
+
+      d_ode_solver.Step(temperature_block, t, dt);
+      {
+         temperature_block_gf.SetFromTrueDofs(temperature_block);
+
+         ParSubMesh::Transfer(temperature_block_gf, cylinder_surface_gf);
+         ParSubMesh::Transfer(cylinder_surface_gf, temperature_cylinder_gf);
+
+         temperature_cylinder_gf.GetTrueDofs(temperature_cylinder);
+      }
+      cd_ode_solver.Step(temperature_cylinder, t, dt);
+
+      if (last_step || (ti % vis_steps) == 0)
+      {
+         if (myid == 0)
+         {
+            out << "step " << ti << ", t = " << t << std::endl;
+         }
+
+         temperature_cylinder_gf.SetFromTrueDofs(temperature_cylinder);
+         temperature_block_gf.SetFromTrueDofs(temperature_block);
+
+         cyl_sol_sock << "solution\n" << cylinder_submesh << temperature_cylinder_gf <<
+                      std::flush;
+         block_sol_sock << "solution\n" << block_submesh << temperature_block_gf <<
+                        std::flush;
```

Comment:
Not exactly sure what goes wrong here, but the visualization seems to be broken. 

![subdomain-broken](https://user-images.githubusercontent.com/9196588/165183283-0a2bd5c5-5a60-43a9-a3f2-2db618a21521.png)


Initial solution just works fine.

Reply:
No idea why this shows up sometimes. Some simulations work 100% fine, other runs are garbled like this. I'm investigating.

Reply:
I often hotfix these by also sending the parallel command again, i.e.

```cpp
         cyl_sol_sock << "parallel " << num_procs << " "
                           << myid << "\n";
         cyl_sol_sock << "solution\n" << cylinder_submesh << temperature_cylinder_gf <<
                      std::flush;

         block_sol_sock << "parallel " <<num_procs << " "
                           << myid << "\n";
         block_sol_sock << "solution\n" << block_submesh << temperature_block_gf <<
                        std::flush;
```
 Howver, I am not sure if this is the intended way to use glvis.